src/codegen/doc/mod.rs

```diff
@@ -517,13 +559,38 @@ where
             }
         }
 
-        if let Some(ref doc) = fn_.ret().doc {
+        let out_parameters: Vec<_> = no_array_length_params
+            .iter()
+            .filter(|param| {
+                param.direction == ParameterDirection::Out
+                    && !(param.name == "error" && param.c_type == "GError**")
+            })
+            .collect();
+
+        if fn_.ret().doc.is_some() || !out_parameters.is_empty() {
             writeln!(w, "\n# Returns\n")?;
```

Comment:
You probably don't need to fold the `if let Some(ref doc) = fn_.ret().doc` and `for parameter in out_parameters` inside this block?

I'm not sure what's easier to read...

Reply:
They are both "return" results... not sure really how to make it any better without over complex-ifying  this PR

Reply:
Taking them outside of the `if fn_.ret().doc.is_some() || !out_parameters.is_empty() {` block as that means the same, and saves a bit of indentation. It doesn't change anything functionally and might, OTOH, make it less readable too...

Reply:
Ah I see what you mean

Reply:
I somehow thought that wouldn't work but now that I think of :P needs more coffee! Thanks for all the great suggestions :-)

Reply:
If you're doing this, do the same for `if let Some(ref doc) = fn_.ret().doc {`.

Reply:
```rust
        if fn_.ret().doc.is_some() || !out_parameters.is_empty() {
            writeln!(w, "\n# Returns\n")?;
        }
```
Is what you'll have in the end ;)